### DISCUSSION TITLE
Comparing current antsMotionCorr with improved version using more sensible smoothing

### DISCUSSION THREAD

**Discussion by Gab-D-G** [CoBrALab/RABIES#233] [Show and tell]
With the current ANTs version, antsMotionCorr smoothing parameters are in mm, and RABIES' default strategy uses -s 1x0, meaning a smoothing of 1mm then no smoothing. With more recent updates, antsMotionCorr smoothing are in voxel instead of mm https://github.com/ANTsX/ANTs/pull/1350, like antsRegistration. I compare here a more sensible smoothing framework with -s 3x2x1x0 in voxels instead of mm with an updated version of ANTs pulled from commit 55040a3065985f601960adf1a1bc34275cee3ec5 https://github.com/ANTsX/ANTs/tree/55040a3065985f601960adf1a1bc34275cee3ec5.

The HMC command for the current version:
```
moving=$1
out=$2
fixed=$3

antsMotionCorr -d 3 -o [${out}/motcorr,${out}/motcorr.nii.gz,${out}/motcorr_avg.nii.gz] -m MI[ ${fixed} , \
                ${moving} , 1 , 20 , regular, 0.2 ] -t Rigid[ 0.25 ] -i 50x20 -s 1x0 -f 2x1 -u 1 -e 1 -l 1 -n 10 -v 1
```
The command for the new version:
```
moving=$1
out=$2
fixed=$3

docker run -it --rm -v $PWD:/work_dir testing_hmc \
antsMotionCorr -d 3 -o [${out}/motcorr,${out}/motcorr.nii.gz,${out}/motcorr_avg.nii.gz] -m MI[ ${fixed} , \
                ${moving} , 1 , 20 , regular, 0.2 ] -t Rigid[ 0.1 ] -i 50x50x50x20 -s 3x2x1x0 -f 4x3x2x1 -u 1 -e 1 -l 1 -n 10 -v 1
```

I am testing here the two versions on 3 awake scans with high EPI contrast and clear motion, and 2 EPI scans from another dataset with lower EPI contrast and less motion.

**Awake scan 1:**
![sub-jgrAesAWc11L_ses-1_task-rest_acq-EPI_run-3_bold_motion_parameters](https://user-images.githubusercontent.com/34081874/176958402-314f1027-dda1-4a13-9615-e7ef705856f4.png)
![sub-jgrAesAWc11L_ses-1_task-rest_acq-EPI_run-3_bold_std](https://user-images.githubusercontent.com/34081874/176959618-dbf71bce-ba85-46ae-9f76-577e523cc448.png)

**Awake scan 2:**
![sub-jgrAesAWc11L_ses-2_task-rest_acq-EPI_run-3_bold_motion_parameters](https://user-images.githubusercontent.com/34081874/176959679-d811842b-1f72-41af-8c8d-4caba80f6f8b.png)
![sub-jgrAesAWc11L_ses-2_task-rest_acq-EPI_run-3_bold_std](https://user-images.githubusercontent.com/34081874/176959693-eb17e036-e638-42fc-b7d3-85c5baf02bae.png)

**Awake scan 3:**
![sub-jgrAesAWc11R1L_ses-1_task-rest_acq-EPI_run-3_bold_motion_parameters](https://user-images.githubusercontent.com/34081874/176958634-12c588ca-c3b3-40f7-b203-e10baac8d09c.png)
![sub-jgrAesAWc11R1L_ses-1_task-rest_acq-EPI_run-3_bold_std](https://user-images.githubusercontent.com/34081874/176959716-66ac8e09-2c59-47b4-860c-95ab37b14828.png)

**Low contrast scan 1:**

![sub-SLC03_ses-1_task-led_run-7_bold_motion_parameters](https://user-images.githubusercontent.com/34081874/176958678-6a8acf4e-aa55-465a-b885-b17e0f3a588b.png)
![sub-SLC03_ses-1_task-led_run-7_bold_std](https://user-images.githubusercontent.com/34081874/176959737-a9325797-5a18-4d40-83c7-9e93a016f765.png)

**Low contrast scan 2:**
![sub-SLC05_ses-2_task-rest_run-5_bold_motion_parameters](https://user-images.githubusercontent.com/34081874/176958717-b76df306-e18a-4f4c-86af-fb56620bbca1.png)
![sub-SLC05_ses-2_task-rest_run-5_bold_std](https://user-images.githubusercontent.com/34081874/176959756-48b46002-337b-4d98-9785-82fd572a405d.png)

**Conclusions:**
In the cases of high motion in the awake scans, the estimated parameters are largely identical across all 6 parameters. In the case of the lower quality scans, the upgraded version tends to make larger estimates for the 2nd and 3rd parameters for both translations and rotations. It is unclear why this is the case, or which of the two estimate is more accurate.
When looking at the standard deviation at each voxel after motion realignment, the standard deviation maps are essentially identical in all cases, suggesting that the differences between the implementations are minimal.

**Comment by gdevenyi** [comment]
A followup here, this option:
```
--metric MI[fixedImage,movingImage,metricWeight,numberOfBins,<samplingStrategy={Regular,Random}>,<samplingPercentage=[0,1]>,<useGradientFilter=false>]
```

the UseGradientFilter option was added.

I would be interested to see how:
```
MI[ ${fixed} ,${moving} , 1 , 20 , regular, 0.2, 1 ]
```

Works
